### ISSUE TITLE
Support column specs for DataTable

### ISSUE THREAD

**Issue by hhaensel** [GenieFramework/StippleUI.jl#52]
With PR #51 I propose to add a field columnspecs to DataTableOptions that makes it possible to provide extra formatting specs for columns.

As there are very many options to columns which are currently not all reflected in the Column struct I decided to go for an option dictionary instead. (My use case is provide formatting to numbers.

Usage is as follows:

```julia
import Stipple.opts

df = DataFrame(a = sin.(-π:π/10:π), b = cos.(-π:π/10:π), c = string.(rand(21)))
dt = DataTable(df)
dt.opts.columnspecs[r"^(a|b)$"] = opts(format = jsfunction(raw"(val, row) => `${100*val.toFixed(3)}%`"))

model.table[] = dt
```

columnspecs is a `Dict{Union{String, Regex}, Dict{Symbol, Any}}` and provides either exact or Regex matching for column selection. So above example will format columns "a" and "b" in percent, all other columns are untouched. Also a column "aa" is not formatted due to the `'^'` and `'$'` characters.
![image](https://user-images.githubusercontent.com/31985040/157465467-fb97e973-d468-4f6b-840e-bf71ca5ff3ea.png)

**Comment by AbhimanyuAryan** [comment]
> No, it's from StippleUI.Tables

Aah OK. Thanks :)

**Comment by hhaensel** [comment]
This is already fixed by #51
